test/tree/tree/test_structural.py

```diff
@@ -273,3 +287,15 @@ class MyTreeValue(TreeValue):
                 MyTreeValue({'v': {'x': 3, 'y': 8}}),
             ]
         }
+
+        nt = namedtuple('nt', ['a', 'b', 'c'])
+        t11 = MyTreeValue({
+            'x': nt(1, 4, 7),
+            'y': nt(2, 5, 8),
+            'z': {'v': nt(3, 6, 9), },
+        })
+        assert rise(t11) == nt(
+            MyTreeValue({'x': 1, 'y': 2, 'z': {'v': 3}}),
+            MyTreeValue({'x': 4, 'y': 5, 'z': {'v': 6}}),
+            MyTreeValue({'x': 7, 'y': 8, 'z': {'v': 9}}),
+        )
```

Comment:
we should also test and check the usage of getattr about namedtuple, such as `nt.a`